src/chunk_constraint.c

```diff
@@ -1301,12 +1303,16 @@ check_chunk_constraint_violated(Oid chunk_relid, const Dimension *dim, const Dim
 		Datum datum;
 		int64 value;
 
-		if (NULL != dim->partitioning)
-			datum = ts_partitioning_func_apply_slot(dim->partitioning, slot, &isnull);
-		else
-			datum = slot_getattr(slot, dim->column_attno, &isnull);
+		datum = slot_getattr(slot, attno, &isnull);
 		Assert(!isnull);
 
+		if (NULL != dim->partitioning)
```

Comment:
i do not like NULL != expression variant this always makes my brain pause cause it feels unnatural to read
```suggestion
		if (dim->partitioning)
```